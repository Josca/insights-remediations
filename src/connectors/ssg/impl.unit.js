'use strict';

const fs = require('fs');
const path = require('path');
const impl = require('./impl');
const base = require('../../test');
const { mockRequest } = require('../testUtils');
const request = require('../../util/request');
const identifiers = require('../../util/identifiers');

describe('ssg impl', function () {
    beforeEach(mockRequest);

    test('returns template', async function () {
        // eslint-disable-next-line security/detect-non-literal-fs-filename
        const data = fs.readFileSync(path.join(__dirname, 'mock', 'standard', 'service_rsyslog_enabled.yml'), 'utf-8');

        const http = base.getSandbox().stub(request, 'run').resolves({
            statusCode: 200,
            body: data,
            headers: {}
        });

        const id = identifiers.parse('ssg:rhel7|standard|xccdf_org.ssgproject.content_rule_sshd_disable_root_login');
        await expect(impl.getTemplate(id)).resolves.toMatchSnapshot();
        http.callCount.should.equal(1);
    });
});
