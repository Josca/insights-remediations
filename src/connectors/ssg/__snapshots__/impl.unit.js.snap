// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ssg impl returns template 1`] = `
"- name: service_rsyslog_enabled
  hosts: '@@HOSTS@@'
  become: true
  tags:
  - enable_strategy
  - CCE-80188-6
  - low_disruption
  - NIST-800-53-AU-12
  - service_rsyslog_enabled
  - low_complexity
  - medium_severity
  - NIST-800-53-AU-4(1)
  tasks:
  - name: Enable service rsyslog
    service:
      name: rsyslog
      enabled: 'yes'
      state: started
    when: ansible_virtualization_role != \\"guest\\" or ansible_virtualization_type !=
      \\"docker\\"
"
`;
